src/Compilers/CSharp/Portable/Binder/Binder.WithQueryLambdaParametersBinder.cs

```diff
@@ -43,14 +43,14 @@ protected override BoundExpression BindRangeVariable(SimpleNameSyntax node, Rang
                         // the range variable maps directly to a use of the parameter of that name
                         var value = base.parameterMap[qv.Name];
                         Debug.Assert(value.Count == 1);
-                        translation = new BoundParameter(node, value.Single()) { WasCompilerGenerated = true };
```

Comment:
This is causing test failures where GetSymbolInfo is now returning parameter symbol instead of range variable symbol. I will revert the changes to this file and attempt to handle this correctly in the operation factory.